website/content/docs/configuration/worker/worker-configuration.mdx

```diff
@@ -0,0 +1,213 @@
+  public_addr = "5.1.23.198"
+}
+```
+
+Workers using the KMS authorization flow also require a KMS block designated for `worker-auth`. This is the KMS configuration for authentication between the workers and controllers and must be present. Example (not safe for production!):
+
+```hcl
+kms "aead" {
+	purpose = "worker-auth"
+	aead_type = "aes-gcm"
+	key = "X+IJMVT6OnsrIR6G/9OTcJSX+lM9FSPN"
+	key_id = "global_worker-auth"
+}
+```
+
+The upstream controller or worker must have a `kms` block that references the
+same key and purpose. If both a controller and worker are running as the same
+server process, only one stanza is needed. 
+
+For Multi-Hop workers, It is also possible to specify a `kms` block with the `downstream-worker-auth` purpose. If specified, this will be a separate KMS that can be used for authenticating new downstream nodes. Blocks with this purpose can be specified multiple times. This allows a single upstream node to authenticate with one key to its own upstream (via the `worker-auth` purpose) and then serve as an authenticating upstream to nodes
+across various networks, each with their own separate KMS system or key:
+
+```hcl
+kms "aead" {
+	purpose = "downstream-worker-auth"
+	aead_type = "aes-gcm"
+	key = "XthZVtFtBD1Bw1XwAWhZKVrIwRhR7HcZ"
+	key_id = "iot-nodes-auth"
+}
+
+kms "aead" {
+	purpose = "downstream-worker-auth"
+	aead_type = "aes-gcm"
+	key = "OLFhJNbEb3umRjdhY15QKNEmNXokY1Iq"
+	key_id = "production-nodes-auth"
+}
+```
+
+In the examples above we are encoding key bytes directly in the configuration
+file. This is because we are using the `aead` method where you directly supply a
+key; in production you'd want to use a KMS such as AWS KMS, GCP CKMS, Azure
+KeyVault, or HashiCorp Vault. For a complete guide to all available KMS types,
+see our [KMS documentation](/boundary/docs/configuration/kms).
+
+## Complete configuration example
+
+```hcl
+listener "tcp" {
+	purpose = "proxy"
+	tls_disable = true
+	address = "127.0.0.1"
+}
+
+worker {
+  # Name attr must be unique across workers
+  name = "demo-worker-1"
+  description = "A default worker created for demonstration"
+
+  # Workers must be able to reach upstreams on :9201
+  initial_upstreams = [
+    "10.0.0.1",
+    "10.0.0.2",
+    "10.0.0.3",
+  ]
+
+  public_addr = "myhost.mycompany.com"
+
+  tags {
+    type   = ["prod", "webservers"]
+    region = ["us-east-1"]
+  }
+}
+
+# must be same key as used on controller config
+kms "aead" {
+	purpose = "worker-auth"
+	aead_type = "aes-gcm"
+	key = "X+IJMVT6OnsrIR6G/9OTcJSX+lM9FSPN"
+	key_id = "global_worker-auth"
+}
+```
+
+[`initial_upstreams`](/boundary/docs/configuration/worker/overview#initial_upstreams)
+are used to connect to upstream Boundary clusters.
+
+## Resources
+
+For more on how `tags{}` in the above configuration are used to facilitate
+routing to the correct target, refer to the [Worker
+Tags](/boundary/docs/concepts/filtering/worker-tags) page.
+
+
+## KMS configuration
```

Comment:
I think this section should be moved/ perhaps become a subsection for the worker led auth, since that's what it applies to.

Reply:
this is a good idea -- I wont include this for the initial update as we do call out that it's only available for worker-led & controller-led, but we should figure out if we want to refactor the docs to more easily include this as a subsection (cc @Dan-Heath)